See
https://www.illumos.org/issues/16523

Patch locally in Tribblix until such time as it gets resolved upstream.

Yes, this isn't the final form of the patch, as it duplicates the definition
on both branches. But this way is more of a hint to the reader that something
has been changed.

--- a/usr/src/uts/common/sys/int_types.h	Wed Mar 27 10:10:19 2024
+++ b/usr/src/uts/common/sys/int_types.h	Wed May 29 20:49:36 2024
@@ -72,7 +72,7 @@
  * signed integral.
  */
 #if defined(_CHAR_IS_SIGNED)
-typedef char			int8_t;
+typedef signed char		int8_t;
 #else
 typedef signed char		int8_t;
 #endif
